### PR TITLE
Sync active channels when no channel list

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Links

https://github.com/GetStream/stream-chat-swiftui/issues/126

### 🎯 Goal

Fix a bug when syncing active channels w/o channel list.

### 📝 Summary

When there's a channel screen shown directly (w/o channel list), the sync endpoint was not called, which lead to losing messages sent in the meantime.

### 🛠 Implementation

If there are no local channel ids to sync, we take the ones from the active channel ids.

### 🧪 Manual Testing Notes

You need an app which directly shows channel. Put the app into background, send messages from another device, come back and the messages are loaded now (previously they weren't).

### ☑️ Contributor Checklist
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

**Review by polqf — COMMENTED** [review]
Looks good! Just missing some tests here

**Review by nuno-vieira — DISMISSED** [review]
LGTM! ✅

File: Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift [hunk 1]
```diff
@@ -138,8 +138,8 @@ final class SyncOperations_Tests: XCTestCase {
 
         operation.startAndWaitForCompletion()
 
-        XCTAssertEqual(context.watchedAndSynchedChannelIds.count, 0)
-        XCTAssertNotCall("recoverWatchedChannel(completion:)", on: controller)
+        XCTAssertEqual(context.watchedAndSynchedChannelIds.count, 1)
+        XCTAssertCall("recoverWatchedChannel(completion:)", on: controller)
```
Comment (nuno-vieira) [inline Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift:142]: Why are we reverting the test here? Although the test passes now, it is not what it is supposed to be testing. Right? @polqf

File: Sources/StreamChat/Repositories/SyncOperations.swift [hunk 2]
```diff
@@ -72,12 +72,6 @@ final class WatchChannelOperation: AsyncOperation {
                 return
             }
 
-            // Reset only if it needs recovery
-            if let cid = controller.cid, context.synchedChannelIds.contains(cid) {
-                done(.continue)
-                return
-            }
```
Comment (nuno-vieira) [inline Sources/StreamChat/Repositories/SyncOperations.swift:79]: Are we sure about this?

File: Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift (hunk 1 above)
Comment (polqf) [reply Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift:142]: This test should be split in 2.
- One that checks that given a channelId that was synched (present in `context.synchedChannelIds `), it calls `recoverWatchedChannel`, but the end result of `context.watchedAndSynchedChannelIds` is 1
- One that checks that given  a channelId that was NOT synched (NOT present in `context.synchedChannelIds `), it calls `recoverWatchedChannel`, and the result of `context.watchedAndSynchedChannelIds` is 1 and `context.synchedChannelIds` keeps being 0

File: Sources/StreamChat/Repositories/SyncOperations.swift (hunk 2 above)
Comment (polqf) [reply Sources/StreamChat/Repositories/SyncOperations.swift:79]: @nuno-vieira this is a fix that's meant to be safe given the error we are fixing, but maybe not the "most performant" as it can lead to unnecessary calls. (1 more call at most)

There is a task in JIRA to review this in depth and prevent that unnecessary call if possible

File: Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift (hunk 1 above)
Comment (martinmitrevski) [reply Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift:142]: @polqf, the current test was already doing the first check. I've added another one that tests the second point.

File: Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift [hunk 3]
```diff
@@ -141,6 +141,21 @@ final class SyncOperations_Tests: XCTestCase {
         XCTAssertEqual(context.watchedAndSynchedChannelIds.count, 1)
         XCTAssertCall("recoverWatchedChannel(completion:)", on: controller)
     }
+    
+    func test_WatchChannelOperation_availableOnRemote_notSynched() {
+        let context = SyncContext(lastSyncAt: .init())
+        let controller = ChatChannelController_Spy(client: client)
+        controller.state = .remoteDataFetched
+        context.localChannelIds.insert(controller.cid!, at: 0)
```
Comment (polqf) [inline Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift:149]: Why is this needed here?

**Review by polqf — COMMENTED** [review]
Just one comment. Rest looks good

File: Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift (hunk 3 above)
Comment (martinmitrevski) [reply Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift]: it's not needed, fixed

**Review by polqf — APPROVED** [review]
